Creating actor stuff adapters and holders

dovahcin opened the pull request.

dovahcin committed:
2 Layouts Created

dovahcin committed:
Created actorlistadapter and converted listadapter
HorizontalListAdapter now can apply two Holders,

dovahcin commented on app/src/main/java/com/movie/android/presentation/features/explore/adapter/HorizontalActorListHolder.kt:
```diff
@@ -0,0 +1,26 @@
+) : RecyclerView.ViewHolder(binding.root) {
+
+    companion object {
+        fun create(parent: ViewGroup, actorClick: () -> Unit) =
```
این پرانتزه باید Actor می داشت

rezaiyan replied:
کلاس Actor  پیدا نمیشه

rezaiyan commented on app/src/main/java/com/movie/android/domain/HorizontalItem.kt:
```diff
@@ -0,0 +1,11 @@
+package com.movie.android.domain
+
+sealed class HorizontalItem(val viewType: Type) {
```
بهتره از همون ExploreItem استفاده کنیم
و ایتم جدید ایجاد نکنیم

فرض کن لیست اکسپلور فقط یک اداپتر قراره داشته باشه و همه ایتم‌های باید از یک مدل باشن

dovahcin committed:
completed actors list
Made Actors request
Made Actors data in Explore Repos
Actors List now functions and can be shown up

dovahcin commented on app/src/main/res/layout/item_actor_horizontal.xml:
```diff
@@ -15,7 +15,7 @@
             android:id="@+id/actorsImageView"
             android:layout_width="120dp"
-            android:layout_height="0dp"
+            android:layout_height="120dp"
```
فک کردم اینطوری دایره ش کامل میشه الان یادم افتاد که باید scaleType می ذاشتم

dovahcin committed:
reformated the item_actor and fixed scalteType

dovahcin committed:
took 15 items for actors list

rezaiyan commented on app/src/main/java/com/movie/android/presentation/features/explore/ExploreViewModel.kt:
```diff
@@ -33,9 +33,9 @@ class ExploreViewModel(
-                    .collect { movies ->
-                        _uiState.value = Success(movies)
-                        state.set(EXPLORE_KEY, Success(movies))
+                    .collect { results ->
```
من معمولا سعی میکنم اسم‌ها رو یکم جنرال تعریف کنم که تو ریفکتور‌ها نیاز به تغییر نباشه :)

rezaiyan commented on app/src/main/java/com/movie/android/presentation/features/explore/adapter/ExploreAdapter.kt:
```diff
@@ -18,27 +20,32 @@ class ExploreAdapter(
             Type.Promotions.ordinal -> TODO()
-            Type.Artists.ordinal -> TODO()
+            Type.Artists.ordinal -> HorizontalActorListHolder.create(parent, actorClick)
             else -> EmptyHolder.create(parent)
```
تو کد‌های بقیه شاید ببینی تو else مقدار null برمیگردونن که اونم درسته
ولی منم مثل شما ویوهلدر خالی رو ترجیح میدم👌

rezaiyan commented on app/src/main/res/layout/item_actor_horizontal.xml:
```diff
@@ -15,7 +15,7 @@
             android:id="@+id/actorsImageView"
             android:layout_width="120dp"
-            android:layout_height="0dp"
+            android:layout_height="120dp"
```
توی گوشی من دایره‌ای نبود!
<img src="https://user-images.githubusercontent.com/16861750/131215965-19b5c274-6d53-4168-8565-998326502115.png" width="20%"/>

dovahcin replied:
Outdated
